tests/storage/export.go

```diff
@@ -1492,6 +1495,283 @@ var _ = SIGDescribe("Export", func() {
+			vm := tests.NewRandomVMWithDataVolumeAndUserDataInStorageClass(
+				cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskCirros),
+				util.NamespaceTestDefault,
+				bashHelloScript,
+				sc)
+			libstorage.AddDataVolumeTemplate(vm, blankDv)
+			addDataVolumeDisk(vm, "blankdisk", blankDv.Name)
+			vm = createVM(vm)
+			snapshot := createAndVerifyVMSnapshot(vm)
+			Expect(snapshot).ToNot(BeNil())
+			defer deleteSnapshot(snapshot)
+
+			// Run vmexport
+			By("Running vmexport command")
+			virtctlCmd := clientcmd.NewRepeatableVirtctlCommand(commandName, "create", vmeName, "--snapshot", snapshot.Name, "--namespace", util.NamespaceTestDefault)
+			err = virtctlCmd()
+			Expect(err).ToNot(HaveOccurred())
+			checkForReadyExport(vmeName)
+		})
+
+		It("Create succeeds using VM source", func() {
+			// Create a populated VM
+			vm := tests.NewRandomVMWithDataVolumeWithRegistryImport(
+				cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine),
+				util.NamespaceTestDefault,
+				sc,
+				k8sv1.ReadWriteOnce)
+			vm.Spec.Running = pointer.BoolPtr(true)
+			vm = createVM(vm)
+			Eventually(func() virtv1.VirtualMachineInstancePhase {
+				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return ""
+				}
+				Expect(err).ToNot(HaveOccurred())
+				return vmi.Status.Phase
+			}, 180*time.Second, time.Second).Should(Equal(virtv1.Running))
+
+			By("Stopping VM, we should get the export ready eventually")
+			vm = stopVM(vm)
+
+			// Run vmexport
+			By("Running vmexport command")
+			virtctlCmd := clientcmd.NewRepeatableVirtctlCommand(commandName, "create", vmeName, "--vm", vm.Name, "--namespace", util.NamespaceTestDefault)
+			err = virtctlCmd()
+			Expect(err).ToNot(HaveOccurred())
+			checkForReadyExport(vmeName)
+		})
+
+		It("Create fails when the vmexport already exists", func() {
+			vmExport := createRunningPVCExport(sc, k8sv1.PersistentVolumeFilesystem)
+			vmeName = vmExport.Name
+			// Run vmexport
+			By("Running vmexport command")
+			virtctlCmd := clientcmd.NewRepeatableVirtctlCommand(commandName, "create", vmeName, "--pvc", vmExport.Spec.Source.Name, "--namespace", util.NamespaceTestDefault)
+			err = virtctlCmd()
+			Expect(err).To(HaveOccurred())
+			errString := fmt.Sprintf("VirtualMachineExport '%s/%s' already exists", vmExport.Namespace, vmeName)
+			Expect(err.Error()).Should(Equal(errString))
+		})
+
+		It("Delete succeeds", func() {
+			vmExport := createRunningPVCExport(sc, k8sv1.PersistentVolumeFilesystem)
+			vmeName = vmExport.Name
+			// Run vmexport
+			By("Running vmexport command")
+			virtctlCmd := clientcmd.NewRepeatableVirtctlCommand(commandName, "delete", vmeName, "--namespace", util.NamespaceTestDefault)
+			err = virtctlCmd()
+			Expect(err).ToNot(HaveOccurred())
+			vmExport, err = virtClient.VirtualMachineExport(util.NamespaceTestDefault).Get(context.Background(), vmExport.Name, metav1.GetOptions{})
```

Comment:
probably should do it in an Eventually since it might take a second for the delete to happen?